### ISSUE TITLE
predict abnormal results of T-net 

### ISSUE THREAD

**Issue by kkkmax** [tsing90/pytorch_semantic_human_matting#4]
- I use data enhancement to generate 3000 + pairs of images and mask images to train T-net. Losses are as follows

average  loss: 0.75172
saving model ....
epoch end, shuffle datasets again ...
[1 / 300] 	loss: 0.69093	time: 1548  average loss: 0.55106
saving model ....   epoch end, shuffle datasets again ...
[2 / 300] 	loss: 0.53391	time: 1538
average loss: 0.47728 saving model ....epoch end, shuffle datasets again ...
[3 / 300] 	loss: 0.46429	time: 1537 average loss: 0.42489
saving model ....epoch end, shuffle datasets again ...
[4 / 300] 	loss: 0.41367	time: 1537 average loss: 0.38102 
saving model .... epoch end, shuffle datasets again ...
---Ellipsis part
[46 / 300] 	loss: 0.07379	time: 1542 
average loss: 0.07126 saving model ....epoch end, shuffle datasets again ...

- When the loss dropped to 0.07, I thought there would be a good image segmentation result, so I predicted that T-net (PSPNet50) did not seem to work properly.
![4](https://user-images.githubusercontent.com/42377340/54809483-5b411580-4cbe-11e9-894c-f7eb45468146.png)
====================================================

- Change the training load data code to：
![4](https://user-images.githubusercontent.com/42377340/54809632-d1de1300-4cbe-11e9-9699-befba092fe44.png)

- In theory, T-Net (PSPNet) will have normal segmentation results, but my results don't look right, but I'm not sure where the problem is.

**Comment by tsing90** [comment]
I've updated the data preparation method, please refer to the readme file for details.

**Comment by kkkmax** [comment]
> * I use data enhancement to generate 3000 + pairs of images and mask images to train T-net. Losses are as follows
> 
> average loss: 0.75172
> saving model ....
> epoch end, shuffle datasets again ...
> [1 / 300] loss: 0.69093	time: 1548 average loss: 0.55106
> saving model .... epoch end, shuffle datasets again ...
> [2 / 300] loss: 0.53391	time: 1538
> average loss: 0.47728 saving model ....epoch end, shuffle datasets again ...
> [3 / 300] loss: 0.46429	time: 1537 average loss: 0.42489
> saving model ....epoch end, shuffle datasets again ...
> [4 / 300] loss: 0.41367	time: 1537 average loss: 0.38102
> saving model .... epoch end, shuffle datasets again ...
> ---Ellipsis part
> [46 / 300] loss: 0.07379	time: 1542
> average loss: 0.07126 saving model ....epoch end, shuffle datasets again ...
> 
> * When the loss dropped to 0.07, I thought there would be a good image segmentation result, so I predicted that T-net (PSPNet50) did not seem to work properly.
>   ![4](https://user-images.githubusercontent.com/42377340/54809483-5b411580-4cbe-11e9-894c-f7eb45468146.png)
>   ====================================================
> * Change the training load data code to：
>   ![4](https://user-images.githubusercontent.com/42377340/54809632-d1de1300-4cbe-11e9-9699-befba092fe44.png)

**Comment by kkkmax** [comment]
> I've updated the data preparation method, please refer to the readme file for details.
@tsing90  thanks you very much!
I prepared the data according to the description of your Readme file, re-prepared the data, 180 DIM images and my own 880 images were all processed into four-channel (RGBA) images, currently training, for training models I want to ask you a few questions:
1) How long have you trained T-Net training? How much do you think the loss will fall to get a robust T-Net model?
2) How long did it take to train M-net, and how long did you train in end-to-end mode?
